### ISSUE TITLE
Async Collections

### ISSUE THREAD

**Issue by maxnowack** [maxnowack/signaldb#397]
The development of Async Collections in SignalDB is aimed at enhancing the database's flexibility and performance, particularly in scenarios where data persistence without in-memory storage is essential. These Async Collections would enable SignalDB to interact with asynchronous storage mechanisms like IndexedDB more efficiently, allowing for data operations without necessarily storing data in the client's memory. This feature is especially relevant for applications that handle large datasets or require extended data persistence beyond the session scope. Furthermore, Async Collections could facilitate the outsourcing of query execution to Web Workers, thereby offloading intensive data processing tasks from the main thread. This approach would significantly improve the application's responsiveness and overall performance, especially in web applications with complex data processing requirements. The introduction of Async Collections in SignalDB aims to provide developers with more options for data management, catering to a broader range of application architectures and performance needs.

**Comment by rougsig** [comment]
This question isn't directly related to that feature, but it touches on a related point. Am I correct in understanding that collections are currently synchronous (NOT async), and persistence writes cannot be awaited from collection operations?

**Comment by maxnowack** [comment]
@rougsig yes that's right. Take a look at the [Core Concepts in the documentation](https://signaldb.js.org/core-concepts/). There are some events you can subscribe to keep track of the persistence state: https://signaldb.js.org/reference/core/collection/#events

**Comment by Mihailoff** [comment]
I'm new here. Evaluating local database options. Maybe something similar to [mongo's writeConcern](https://www.mongodb.com/docs/manual/reference/write-concern/) concept?

```JS
// Guarantee write persistence before returning postId
const postId = await posts.insert({ title: 'Foo', text: 'Lorem ipsum …' }, { writeConcern: true })
```

```JS
const posts = new Collection({
  persistence: createLocalStorageAdapter('posts'),
  writeConcern: {  // Guarantee write persistence by default at call time
    default: true,
    timeout: 1000
  }
})
```

 Alternatively, 

```JS
posts.on('persistence.error', () => {
  // Can I retry or rollback here?
})
```

**Comment by maxnowack** [comment]
I close this this issue as it is addressed in [v2](#1954) and was already resolved in the latest beta
